central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -523,7 +542,116 @@ func (ds *datastoreImpl) RemovePlopsByPod(ctx context.Context, id string) error
+		log.Errorf("failed to prune process listening on ports by deployment: %v", err)
+	}
+
+	// Delete processes listening on ports orphaned due to missing pods.
+	if _, err := ds.pool.Exec(ctx, deleteOrphanedPLOPPodsWithPodUID); err != nil {
+		log.Errorf("failed to prune process listening on ports by pods: %v", err)
+	}
+
+	return commandTag.RowsAffected()
+}
+
+// PruneOrphanedPLOPsByProcessIndicators prunes PLOPs that match process indicators without pods
+func (ds *datastoreImpl) PruneOrphanedPLOPsByProcessIndicators(ctx context.Context, orphanWindow time.Duration) {
+	// TODO(ROX-22443): Once it is guaranteed that all listening endpoints have PodUIDs, remove this function
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	// Delete processes listening on ports orphaned because process indicators are orphaned due to
+	// missing deployments
+	query := fmt.Sprintf(deleteOrphanedPLOPDeploymentsAndPI, int(orphanWindow.Minutes()))
+	if _, err := ds.pool.Exec(ctx, query); err != nil {
+		log.Errorf("failed to prune process listening on ports by deployment: %v", err)
+	}
+
+	// Delete processes listening on ports orphaned because process indicators are orphaned due to
+	// missing pods.
+	query = fmt.Sprintf(deleteOrphanedPLOPPods, int(orphanWindow.Minutes()))
+	if _, err := ds.pool.Exec(ctx, query); err != nil {
+		log.Errorf("failed to prune process listening on ports by pods: %v", err)
+	}
+}
+
+func (ds *datastoreImpl) readRowsToFindPLOPsWithNoProcessInformation(rows pgx.Rows) ([]string, error) {
+	var ids []string
+
+	for rows.Next() {
+		var serialized []byte
+
+		if err := rows.Scan(&serialized); err != nil {
+			return nil, pgutils.ErrNilIfNoRows(err)
+		}
+
+		var msg storage.ProcessListeningOnPortStorage
+		if err := proto.Unmarshal(serialized, &msg); err != nil {
+			return nil, err
+		}
+
+		process := msg.GetProcess()
+
+		if process == nil {
+			ids = append(ids, msg.GetId())
+		}
+	}
+	return ids, nil
+}
+
+// First gets PLOPs with no matching process indicators and then checks the
+// serialized data to check for process information.
+func (ds *datastoreImpl) getPLOPsToDelete(ctx context.Context) ([]string, error) {
+	rows, err := ds.pool.Query(ctx, getPotentiallyOrphanedPLOPs)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	return ds.readRowsToFindPLOPsWithNoProcessInformation(rows)
+}
+
+// RemovePLOPsWithoutProcessIndicatorOrProcessInfo Finds PLOPs without a matching process indicator and no
+// process information and deletes them. PLOPs can have no matching process indicators, but such PLOPs need
+// to have process information.
+func (ds *datastoreImpl) RemovePLOPsWithoutProcessIndicatorOrProcessInfo(ctx context.Context) (int64, error) {
+
+	plopsToDelete, err := ds.getPLOPsToDelete(ctx)
+
+	if err != nil {
+		return 0, err
+	}
+
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	err = ds.storage.DeleteMany(ctx, plopsToDelete)
+	numRecordsToDelete := len(plopsToDelete)
```

Comment:
No need for the intermediate var. As a general pattern, there typically shouldn't be a space nor any code between and err being returned and checking its err value. e.g.

```
err := result()
if err != nil {
  return err
}
do something assuming that err was nil
```

Reply:
Done